documentation/git.md

```diff
@@ -0,0 +1,144 @@
+
+### Add the task identifier
+
+Add the id of the appropriate sub-task to help locate specifications and discussions related to the work done in the branch.
+
+From here on, an underscore ‘-’ is the separator.
+
+### Express the work to be done
+
+Represent the contents of the branch in a few English words. Derive these from the name of the product backlog item (PBI) to which the sub-task belongs, or in the case of larger PBIs, from the sub-task itself.
+
+### Examples
+
+All together, good branch names look like this:
+
+- `feature/68178-introduce-checkbox`
+- `bug/70447-correct-button-outline`
+- `refactor/70368-use-singular-for-enums`
+- `task/82104-add-initial-git-guidelines`
+
+## Describe each commit
+
+Take some time to determine a good description for your commit. Try to describe the goal of the change. Don’t describe what also be deduced from the diff.
+
+Provide a summary, in English, using the imperative mood. If necessary, write context or details in prose after an empty line. Read more in [this classic article](https://cbea.ms/git-commit/).
+
+The goal is that anyone can quickly scan the progress of a branch based on the commit messages.
+
+## Tidy up your branch
+
+The commits in your branch should reflect the conceptual steps of the implementation process. This allows reviewers to interpret the changes piece by piece rather than all of them as a whole.
+
+If you commit regularly, this often happens automatically, but you may have committed too often or attempted multiple strategies to reach the end result.
+
+With an ‘interactive rebase’, you can locally improve the clarity of your branch - that is, merge, split or reorder commits. This hides unimportant steps or changes in strategy from the reviewer.
+
+More explanation, for example, [here at SourceTree](https://www.atlassian.com/blog/sourcetree/interactive-rebase-sourcetree).
+
+## Keep up with the main branch
+
+Before making a pull request, ensure your branch is as close to the main one as possible to avoid unnecessary merge conflicts.
+
+First, ensure that your local main branch is current: `git checkout main && git pull && git checkout -`.
+
+### Rebase
+
+This can be done with a rebase: `git rebase main`.
+
+It applies every commit from your branch on top of the main branch. If successful, the entire branch will appear to have started on the most recent state of the main branch.
+
+There may be merge conflicts. As rebasing handles commits individually, one merge conflict may need to be resolved multiple times. After resolving a conflict, do `git rebase --continue`. If you get stuck, roll back everything with `git rebase --abort`.
+
+If this branch already exists on the remote, you must force-push the rebase, overwriting the remote branch with your local one. This can be done via `git push origin --force-with-lease`.
+
+If others have this branch locally, you should first consult with them, as they may lose work if they pull the remote branch after you’ve changed it.
```

Comment:
Let’s agree to disagree here. I still vastly prefer rebasing. If you know how it works, it’s not that error-prone.

We’re not recommending this (“Neither approach is mandatory or prohibited”).

Resolving merge conflicts multiple times is tedious, but we have other guidelines to prevent conflicts from happening in the first place. We advise keeping branches short-lived and updating the branch with `main` often.

As for losing work, it appears that the ‘with lease’ part of the flag aims to prevent just that – [source](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force-with-leaseltrefnamegt). I’ll update the wording to reflect that.